### ISSUE TITLE
Wrong encoding after adding from dataset with different SpecificCharacterSet 

### ISSUE THREAD

**Issue by hczedik** [dcm4che/dcm4che#654]
**Describe the bug**
Adding values from a dataset with a different SpecificCharacterSet does not maintain the Strings correctly.

**To Reproduce**
Unit test:
```
    @Test
    public void testAddWithDifferentCharacterSet() throws Exception {
        String name = "\u00c4neas^R\u00fcdiger";

        Attributes a = new Attributes();
        a.setSpecificCharacterSet("ISO_IR 100");
        a.setString(Tag.PatientName, VR.PN, name);

        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        try (DicomOutputStream out = new DicomOutputStream(baos, UID.ExplicitVRLittleEndian)) {
            out.writeDataset(null, a);
        }
        try (DicomInputStream in = new DicomInputStream(new ByteArrayInputStream(baos.toByteArray()),
                UID.ExplicitVRLittleEndian)) {
            a = in.readDataset(-1, -1);
        }

        Attributes b = new Attributes();
        b.setSpecificCharacterSet("ISO_IR 192");
        b.addSelected(a, Tag.PatientName);

        assertEquals(name, b.getString(Tag.PatientName));
    }
```

Fails with: `org.junit.ComparisonFailure: expected:<[Äneas^Rü]diger> but was:<[�neas^R�]diger>`

- Version 5.21.0

This is a regression caused by the fix for https://github.com/dcm4che/dcm4che/issues/651

**Comment by gunterze** [comment]
The questions is, if the client should not be responsible for adjusting the character sets of the data set before applying the merge. Because, just inserting the decoded string would only work, if all characters of the source character set can be encoded with the destination character set, as in above test case. Applying a necessary adjustment implicitly by the merge operation may lead to unexpected results, detected (possibly long time) after the operation was performed. So a fast fail behavior on a mismatch of the character sets between source and destination data set may be more appropriate.

**Comment by gunterze** [comment]
Solved by  #675

**Comment by hczedik** [comment]
Thanks @gunterze !
